### PR TITLE
Edit variant name on variant edit/create

### LINKED ISSUE #2000: Product variants get non-readable names for humans that seem to be Base64 encoded values
### What I'm trying to achieve
Get, set and display human readable names for the variants of a product.

### Steps to reproduce the problem
In the dashboard, create a product and then create variants for it.

### What I expected to happen
A human readable name could be set for each variant of a product. At least, the dashboard should allow editing the product variant name.
Instead of that, a Base 64 encoded value is saved in the "name" column of the "product_productvariant" database table. Later, the dashboard web UI renders it as read only and [a store instance based on the `react-storefront` project](https://github.com/saleor/react-storefront) also renders those unreadable names to the end user.

### Screenshots
product-variant-ugly-names

product-variants-ugly-names-sql-details

react-storefront-renders-ugly-product-variant-names

**System information**

- Saleor dashboard version 3.1.5
- Saleor core version 3.1.9

Operating system: 
- [ ] Windows
- [x] Linux
- [ ] MacOS
- [ ] Other 

Browser:
- [x] Firefox
- [x] Chrome
- [ ] Edge
- [ ] Other

### PR DESCRIPTION
**Pull request by krzysztofzuraw**
I want to merge this change to fix #2000. I've added:
* required text input to change the name for both edit & create variant forms
* logic to change `Variant Name` based on `Variant Selection Attributes`
* validation logic on frontend side (the same pattern as in product create)

## Screenshots

Before
---
CleanShot 2022-10-17 at 14 07 50@2x
CleanShot 2022-10-17 at 14 08 15@2x

After
---
image
image
image

### Pull Request Checklist

1. [x] This code contains UI changes
2. [x] All visible strings are translated with proper context including data-formatting
3. [x] Attributes `[data-test-id]` are added for new elements
4. [ ] Changes are mentioned in the changelog
5. [x] The changes are tested in different browsers and in light/dark mode

### Test environment config

API_URI=https://automation-dashboard.staging.saleor.cloud/graphql/
MARKETPLACE_URL=https://marketplace-gray.vercel.app/

### Do you want to run more stable tests?
To run all tests, just select the stable checkbox. To speed up tests, increase the number of containers. Tests will be re-run only when the "run e2e" label is added.

1. [ ] stable
2. [ ] giftCard
3. [ ] category
4. [ ] collection
5. [ ] attribute
6. [ ] productType
7. [ ] shipping
8. [ ] customer
9. [ ] permissions
10. [ ] menuNavigation
11. [ ] pages
12. [ ] sales
13. [ ] vouchers
14. [ ] homePage
15. [ ] login
16. [ ] orders
17. [ ] products
18. [ ] app
19. [ ] plugins
20. [ ] translations
21. [ ] navigation

CONTAINERS=1

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-6587 Edit variant name](https://app.clickup.com/t/2549495/SALEOR-6587)

**Review by Droniu — APPROVED** [review]

**Review by Droniu — APPROVED** [review]

File: src/products/components/ProductVariantName/ProductVariantName.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import { useIntl } from "react-intl";
+
+interface Props {
+  value: string;
+  onChange: FormChange<any>;
+  disabled?: boolean;
+  errors: ProductErrorFragment[];
+}
+
+const ProductVariantName = ({ value, onChange, disabled, errors }: Props) => {
```
Comment (andrzejewsky) [inline src/products/components/ProductVariantName/ProductVariantName.tsx:17]: please use the same convention to define prop types, everywhere else we use FC

File: src/products/components/ProductVariantCreatePage/form.tsx [hunk 2]
```diff
@@ -185,10 +193,13 @@ function useProductVariantCreateForm(
...
   const changeMetadata = makeMetadataChangeHandler(handleChange);
-  const handleAttributeChange = createAttributeChangeHandler(
-    attributes.change,
-    triggerChange,
-  );
+
+  const handleAttributeChangeWithName = (id: string, value: string) => {
+    triggerChange();
+    attributes.change(id, value === "" ? [] : [value]);
+    handleChange({ target: { value, name: "name" } });
```
Comment (andrzejewsky) [inline src/products/components/ProductVariantCreatePage/form.tsx:200]: just to comment: don't hesitate to refactor big components. You can spot many of them in our codebase, but they can be decoupled into smaller units where one component/hook means one responsibility. If you feel you can do it, please do - it always makes code more testable and readable

**Review by andrzejewsky — APPROVED** [review]

**Review by Droniu — DISMISSED** [review]

**Review by andrzejewsky — DISMISSED** [review]

**Review by eCzerniak — DISMISSED** [review]

**Review by eCzerniak — DISMISSED** [review]

**Review by andrzejewsky — APPROVED** [review]

**Review by eCzerniak — APPROVED** [review]

File: src/products/components/ProductVariantName/ProductVariantName.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
+import { Card, CardContent, TextField } from "@material-ui/core";
+import CardTitle from "@saleor/components/CardTitle";
+import { ProductErrorFragment } from "@saleor/graphql";
+import { FormChange } from "@saleor/hooks/useForm";
+import { commonMessages } from "@saleor/intl";
+import { getFormErrors, getProductErrorMessage } from "@saleor/utils/errors";
+import React from "react";
+import { useIntl } from "react-intl";
+
+interface Props {
```
Comment (Droniu) [inline src/products/components/ProductVariantName/ProductVariantName.tsx:10]:
```suggestion
interface ProductVariantNameProps {
```

File: src/products/components/ProductVariantName/ProductVariantName.tsx [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  value: string;
+  onChange: FormChange<any>;
+  disabled?: boolean;
+  errors: ProductErrorFragment[];
+}
+
+const ProductVariantName: React.FC<Props> = ({
```
Comment (Droniu) [inline src/products/components/ProductVariantName/ProductVariantName.tsx:17]:
```suggestion
const ProductVariantName: React.FC<ProductVariantNameProps> = ({
```

File: src/products/utils/validation.ts [hunk 5]
```diff
@@ -34,3 +36,14 @@ export const validateProductCreateData = (data: ProductCreateData) => {
...
+export const validateVariantData = (
+  data: ProductVariantCreateData | ProductVariantUpdateSubmitData,
+) => {
+  let errors: ProductErrorWithAttributesFragment[] = [];
+
+  if (!data.name) {
+    errors = [...errors, createEmptyRequiredError("name")];
+  }
+  return errors;
+};
```
Comment (Droniu) [inline src/products/utils/validation.ts:49]: I don't get it, why are you spreading an empty array?
```suggestion
export const validateVariantData = (
  data: ProductVariantCreateData | ProductVariantUpdateSubmitData,
): ProductErrorWithAttributesFragment[] => (
  !data.name ? [createEmptyRequiredError("name")] : [];
);
```

**Review by Droniu — CHANGES_REQUESTED** [review]

**Review by eCzerniak — APPROVED** [review]

**Review by andrzejewsky — APPROVED** [review]

**Pull request merged**